Patch for bug 289047, taken from Fedora's backintime-0.9.26-3.fc10.src.rpm.

diff -Naur backintime-0.9.26.orig/common/snapshots.py backintime-0.9.26/common/snapshots.py
--- backintime-0.9.26.orig/common/snapshots.py	2009-05-18 10:24:21.000000000 +0200
+++ backintime-0.9.26/common/snapshots.py	2009-09-02 21:17:30.000000000 +0200
@@ -314,7 +314,7 @@
 			return
 
 		path = self.get_snapshot_path( snapshot_id )
-		cmd = "chmod -R a+rwx \"%s\"" %  path
+		cmd = "find \"%s\" -type d -exec chmod u+wx {} \\;" % path
 		self._execute( cmd )
 		cmd = "rm -rfv \"%s\"" % path
 		self._execute( cmd )
@@ -574,8 +574,8 @@
 		new_snapshot_path = self.get_snapshot_path( new_snapshot_id )
 
 		if os.path.exists( new_snapshot_path ):
-			#self._execute( "find \"%s\" -type d -exec chmod +w {} \;" % new_snapshot_path )
-			self._execute( "chmod -R a+rwx \"%s\"" %  new_snapshot_path )
+			self._execute( "find \"%s\" -type d -exec chmod u+wx {} \\;" % new_snapshot_path )
+			#self._execute( "chmod -R a+rwx \"%s\"" %  new_snapshot_path )
 			self._execute( "rm -rf \"%s\"" % new_snapshot_path )
 		
 			if os.path.exists( new_snapshot_path ):
